extension/ballastextension/config_test.go

```diff
@@ -45,6 +45,7 @@ func TestLoadConfig(t *testing.T) {
 		&Config{
 			ExtensionSettings: config.NewExtensionSettings(config.NewIDWithName(typeStr, "1")),
 			SizeMiB:           123,
+			SizeInPercentage:  20,
```

Comment:
I think we had a problem with this, was supported only on some platforms/environments. Is that still the case? If yes then we need to document it. Maybe I miss remember, but please double check.

Reply:
I tested it on mac, windows, linux, docker and k8s and no issue found. 

To get `total memory` used for the percentage calculation first by `cgroup`(for linux*), if cgroup is not set or not supported then we'll get it by `gopsutils` which is cross-platform supported.